### ISSUE TITLE
Feat - store metadata in project file

### ISSUE THREAD

**Issue by PowerUser64** [BespokeSynth/BespokeSynth#1325]
## Summary

Sometimes it can be nice to know when a project was created or when it was last updated. Filesystems are supposed to keep track of this for us, but the values on the filesystem may not always reflect the actual date a project was created or modified if it is copied between locations or downloaded from the internet.

Here are some examples of metadata that could be added:

1. Project creation date
2. Project most recently saved date
3. Version of bespoke project was most recently saved with

### Motivation

This would make it easier for users to keep track of information about their projects. This information could also be used to help prevent users from mistakenly opening projects with versions of bespoke older than the version they were created with.

**Comment by merkys** [comment]
Agree, at least the version of BespokeSynth would be beneficial. This way savestates failing to open would be easier to debug.
